Quantum-Control-Applications/Superconducting/Single Fixed Transmon/rb_without_singleshot_readout.py

```diff
@@ -0,0 +1,258 @@
+                play("x270", "qubit")
+            with case_(12):
+                play("x90", "qubit")
+            with case_(13):
+                play("x270", "qubit")
+            with case_(14):
+                play("y90", "qubit")
+            with case_(15):
+                play("y270", "qubit")
+            with case_(16):
+                play("x270", "qubit")
+                play("y90", "qubit")
+                play("x90", "qubit")
+            with case_(17):
+                play("x270", "qubit")
+                play("y270", "qubit")
+                play("x90", "qubit")
+            with case_(18):
+                play("x180", "qubit")
+                play("y90", "qubit")
+            with case_(19):
+                play("x180", "qubit")
+                play("y270", "qubit")
+            with case_(20):
+                play("y180", "qubit")
+                play("x90", "qubit")
+            with case_(21):
+                play("y180", "qubit")
+                play("x270", "qubit")
+            with case_(22):
+                play("x90", "qubit")
+                play("y90", "qubit")
+                play("x90", "qubit")
+            with case_(23):
+                play("x270", "qubit")
+                play("y90", "qubit")
+                play("x270", "qubit")
+
+
+with program() as rb:
+    depth = declare(int)
+    saved_gate = declare(int)
+    m = declare(int)
+    n = declare(int)
+    n_st = declare_stream()
+    I = declare(fixed)
+    Q = declare(fixed)
+    I_st = declare_stream()
+    Q_st = declare_stream()
+
+    with for_(n, 0, n < n_avg, n + 1):
+        with for_(m, 0, m < num_of_sequences, m + 1):
+            sequence_list, inv_gate_list = generate_sequence()
+
+            with for_(depth, 1, depth <= max_circuit_depth, depth + delta_depth):
+                # Replacing the last gate in the sequence with the sequence's inverse gate
+                # The original gate is saved in 'saved_gate' and is being restored at the end
+                assign(saved_gate, sequence_list[depth])
+                assign(sequence_list[depth], inv_gate_list[depth - 1])
+
+                wait(cooldown_time, "resonator")
+
+                align("resonator", "qubit")
+
+                play_sequence(sequence_list, depth)
+                align("qubit", "resonator")
+                measure(
+                    "readout",
+                    "resonator",
+                    None,
+                    dual_demod.full("cos", "out1", "sin", "out2", I),
+                    dual_demod.full("minus_sin", "out1", "cos", "out2", Q),
+                )
+
+                save(I, I_st)
+                save(Q, Q_st)
+
+                assign(sequence_list[depth], saved_gate)
+        save(n, n_st)
+
+    with stream_processing():
+        I_st.buffer(max_circuit_depth).buffer(num_of_sequences).average().save("I")
+        Q_st.buffer(max_circuit_depth).buffer(num_of_sequences).average().save("Q")
```

Comment:
This `I_st.buffer(max_circuit_depth, num_of_sequences).average().save("I")`  would be equivalent right ?

Reply:
Alright